elastic/kibana
# [Composable template] Create / Edit wizard

This PR adds the Creation and Edition UI flows for the composable index templates.

## How to test

Start by creating a couple of component templates in Dev tools

```sh
PUT /_component_template/base-logs
{
  "version": 1,
   "template": {
    "mappings": {
      "properties": {
       "title": { "type": "text"}
      }
    }
  }
}

PUT /_component_template/base-metrics
{
  "version": 1,
   "template": {
    "mappings": {
      "properties": {
       "myField": { "type": "float"}
      }
    },
    "aliases": {
      "someAlias": {
        "index_routing": "1"
      }
    }
  }
}
```

Now navigate to the index management > Index templates (tab)

* Click on the "Create template" button
* Add a "name" and "Index pattern" in the logistics step
* There should be a "Merge priority" field and a "_meta" field, unique to the composable index template. Priority should only accept numeric values, and _meta any valid JSON object.
* Click "Next" to navigate to the components
* The "base-logs" and "base-metrics" components should appear. You can click the one you want to add. If you add both, you should be able to reorder them.
* Each component has an eye buton icon to be able to view the component inside a flyout.
* You should be able to search for a component.
* You should be able to filter component that
  * have mappings
  * have index settings
  * have aliases

In our example, filtering only with index settings should return no results as we haven't added any index settings to our components.

The next 3 steps are the same as with legacy index templates. Add any index settings, mappings or aliases as you want.

* In the "Review" steo, under the "Request" step, the API endpoint should start with "PUT _index_template/..." (in contrast with legacy index templates whose API is "PUT _template/...")

* Click "Save template". The template should save and you should be redirected to the index templates list page.
* There should be an "edit" icon in the table that lets you edit the template. Go ahead and edit the template making any modification to the template. To verify that the modification have been saved, edit once again the template and check the changed made were persisted.

### Test that legacy index templates work as before

* Click "Create legacy index template" button
* In the "logistics" step, there should be an "Merge order" field (but no "Merge priority" field). And no "_meta" field.
* There should not be a "Components" step
* In the "Review" > "Request" tab, the API endpoint should be "PUT _template/<template-name>"


<img width="1206" alt="Screenshot 2020-06-29 at 12 31 54" src="https://user-images.githubusercontent.com/2854616/85996870-1485b380-ba09-11ea-9370-88ae48fb52b7.png">

<img width="1218" alt="Screenshot 2020-06-29 at 12 44 44" src="https://user-images.githubusercontent.com/2854616/85996884-1780a400-ba09-11ea-8355-e0eaff12881f.png">
